File: advent_of_code/2024/01/first.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        string filePath = "input.txt";
        int ans = 0;
        List<int> l1 = new List<int>();
        List<int> l2 = new List<int>();

        try
        {
            // Read all lines from the file into a string array
            string[] lines = File.ReadAllLines(filePath);

            // Print each line
            foreach (string line in lines)
            {
                int a, b;
                string[] parts = line.Split("   ");
                a = int.Parse(parts[0]);
                l1.Add(a);
                b = int.Parse(parts[1]);
                l2.Add(b);
            }
            l1.Sort();
            l2.Sort();
            for (int i = 0; i < l1.Count; i++)
            {
                ans += Math.Abs(l1[i] - l2[i]);
            }
        }
        catch (Exception e)
        {
            Console.WriteLine("The file could not be read:");
            Console.WriteLine(e.Message);
        }
        Console.WriteLine(ans);
    }
}

File: advent_of_code/2025/04/first.js
const fs = require('node:fs')

const input = fs.readFileSync('./input.txt', 'utf8')

const lines = input.split('\n')

const map = lines.map(line => line.split(''))

let ans = 0

for (let i = 0; i < map.length; i++) {
  for (let j = 0; j < map[i].length; j++) {
    if (map[i][j] === '@') {
      let count = 0
      for (let p = -1; p <= 1; p++) {
        for (let q = -1; q <= 1; q++) {
          if (
            (p === 0 && q === 0)
            || i + p < 0
            || i + p >= map.length
            || j + q < 0
            || j + q >= map[i].length
          ) {
            continue
          }
          if (map[i + p][j + q] === '@') {
            count += 1
          }
        }
      }
      if (count < 4) {
        ans += 1
      }
    }
  }
}

console.log(ans)